### ISSUE TITLE
Regression: Error running pipeline with file dependencies

### ISSUE THREAD

**Issue by annhov** [elyra-ai/elyra#2978]
**Describe the issue**
Kubeflow pipeline fails to run when notebook and its file dependencies are in different directories. This wasn't an issue in version 3.11 as it was possible to set the full path of the file dependency. Now using "Browse" option only allows to set the path from the project root, which adds the redundant part to the final file path (see duplicate 'ml-sample' in the path from logs). The location of .pipeline file does not affect this behavior. 

**To Reproduce**
1. Create an empty directory (project root).
1. Create two folders in it. 
2. Add one json(or other) file in the first folder and one jupyter notebook in the second folder. 
3. Create a new pipeline in the root directory. 
4. Add jupyter notebook from step 2 into the pipeline. 
5. Add json file as a file dependency. 

**Screenshots or log output**
```
{
  "severity": 1,
  "source": "Elyra Pipeline Validation Service",
  "type": "invalidFilePath",
  "message": "Property has an invalid path to a file/dir or the file/dir does not exist.",
  "data": {
    "nodeID": "76691d7e-a471-47a6-8734-b621ae9eae16",
    "nodeName": "read_csv_register_dataset.ipynb",
    "propertyName": "dependencies",
    "value": "/home/jovyan/ml-sample/pipeline-01/ml-sample/jsons/file.json"
  }
}
{
  "severity": 1,
  "source": "Elyra Pipeline Validation Service",
  "type": "invalidFilePath",
  "message": "Property has an invalid path to a file/dir or the file/dir does not exist.",
  "data": {
    "nodeID": "9452e49f-f960-403d-8688-ba3a5ae605c4",
    "nodeName": "TF.ipynb",
    "propertyName": "dependencies",
    "value": "/home/jovyan/ml-sample/pipeline-01/ml-sample/jsons/file.json"
  }
}
```

**Expected behavior**
Ability to modify file dependency path after browsing. 

**Deployment information**
 - Elyra version: 3.13.0.dev0
 - Operating system: linux
 - Installation source: from source
 - Deployment type: Kubeflow [notebook server]

**Pipeline runtime environment**
 - Kubeflow Pipelines

**Comment by akchinSTC** [comment]
@annhov - hi again. so I've been re-familiarizing myself with this part of elyra. While I was able to recreate the issue experienced, I was unable to actually get a valid pipeline running with the described working version (v3.11.0) when specifying an absolute path. IIRC, when we encountered this issue in the past, we suggest users create a symlink to the required dependency in the directory where the notebook resides.

      Project Root        /  Directory A  /    [Notebook A]
                                                [Symlink] ------------------|   
      Project Root        /   Directory B   /    [File A]<------------------|
